### PR TITLE
link and make more consistent uses of Catalog, Item, and Collection

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):** #105

**Proposed Changes:**

1. cleanup text to make uses of Item, Collection, and Catalog more consistent. 

**PR Checklist:**

- [X] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [X] This PR has **no** breaking changes.
- [X] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-api-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -54,5 +54,5 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 See the [stac-spec CHANGELOG](https://github.com/radiantearth/stac-spec/blob/v0.9.0/CHANGELOG.md)
 for STAC API releases prior to or equal to version 0.9.0.
 
-[Unreleased]: <https://github.com/radiantearth/stac-api-spec/compare/master...dev>
```
Comment (m-mohr) [inline CHANGELOG.md:57]: Should still be upper-case as the heading above is also upper-case (first letter)

File: .circleci/rc.yaml [hunk 2]
```diff
@@ -40,4 +41,13 @@ plugins:
     - space 
 # Tables
   - remark-lint-table-pipes
-  - remark-lint-no-literal-urls
```
Comment (m-mohr) [inline .circleci/rc.yaml:43]: Could you give some details on these changes? The remark rules on all STAC repos are aligned and to give predicatable behavior I'd suggest to keep them as before...

File: ogcapi-features/openapi.yaml [hunk 3]
```diff
@@ -31,7 +31,7 @@ paths:
             (link relations `service-desc` and `service-doc`),
             the Conformance declaration (path `/conformance`,
             link relation `conformance`), and the Feature
-            Collections (path `/collections`, link relation
```
Comment (m-mohr) [inline ogcapi-features/openapi.yaml:34]: Maybe also check whether the original in ogcapi-features has this issue and fix it there, too... Makes alignment in the future easier for us...

File: ogcapi-features/openapi.yaml [hunk 4]
```diff
@@ -97,7 +97,7 @@ paths:
         - Features
       summary: the feature collections in the dataset
       description: |-
-        A body of Feature Collections that belong or are used together with additional links.
```
Comment (m-mohr) [inline ogcapi-features/openapi.yaml:100]: I'd rather stick to the original. In OGC API Features they don't use the appended "object" to entities.

**Review by m-mohr — COMMENTED** [review]
Overall LGTM. Added a couple of comments regarding alignment with STAC/OGC.

File: .circleci/rc.yaml (hunk 2 above)
Comment (philvarner) [reply .circleci/rc.yaml:43]: This one was apparently out of sync with the one in stac-spec, which is why it was giving so many errors. I updated it with that one and the rules still work fine.

File: CHANGELOG.md (hunk 1 above)
Comment (philvarner) [reply CHANGELOG.md]: this was a remark warning, reverted now.

File: ogcapi-features/openapi.yaml (hunk 3 above)
Comment (philvarner) [reply ogcapi-features/openapi.yaml:34]: reverted this and the other one in ogcapi

File: .circleci/rc.yaml (hunk 2 above)
Comment (m-mohr) [reply .circleci/rc.yaml:43]: Ah, maybe we did mess up with dev/master here? Not sure, I thought I had updated all repos already.

**Review by m-mohr — APPROVED** [review]

**Pull request merged**